runtime/lua/vim/lsp/handlers.lua

```diff
@@ -369,15 +369,16 @@ function M.hover(_, result, ctx, config)
     end
     return
   end
-  local markdown_lines = util.convert_input_to_markdown_lines(result.contents)
-  markdown_lines = util.trim_empty_lines(markdown_lines)
-  if vim.tbl_isempty(markdown_lines) then
+  local contents, format = util.convert_input_to_render_content(result.contents)
+  contents = util.trim_empty_lines(contents)
+  if vim.tbl_isempty(contents) then
```

Comment:
I think we should not do any pre processing if `result.contents.kind` is `plaintext`. If that's the case then skip `util.convert_input_to_markdown_lines` and just use `result.contents.value` directly.

This also means we don't need to rename `util.convert_input_to_markdown_lines`.

Reply:
I think you're right. Although if the input to `convert_input_to_markdown_lines` is `plaintext` `MarkupContent` then it will just return the unprocessed value. I'll update the docs to specify that.

Reply:
Still, control flow is easier to reason about when the function is only applied to content that _should_ be normalized.

Reply:
@clason I don't disagree with that. I'm just thinking of how this function is part of the public API and we're now changing its contract. Internally (inside this repo) I do agree we should just use it for true Markdown input.

Reply:
Yes, the question of refactoring/deprecating the _external_ API is different (and I'd leave it to @gpanders to take a sledgehammer to it ;))

In the context of this PR, I am _only_ concerned about the internal codepath for showing LSP content (hover, signature, preview).